clang/test/CIR/CodeGen/attribute-annotate-multiple.cpp

```diff
@@ -0,0 +1,169 @@
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-cir -mmlir --mlir-print-ir-before=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s -check-prefix=BEFORE
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-cir -mmlir --mlir-print-ir-after=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s -check-prefix=AFTER
+// RUN: %clang_cc1 -triple aarch64-none-linux-android21 -fclangir -emit-llvm %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll %s -check-prefix=LLVM
+double *a __attribute__((annotate("withargs", "21", 12 )));
+int *b __attribute__((annotate("withargs", "21", 12 )));
+void *c __attribute__((annotate("noargvar")));
+void foo(int i) __attribute__((annotate("noargfunc"))) {
+}
+// redeclare with more annotate
+void foo(int i) __attribute__((annotate("withargfunc", "os", 23 )));
+void bar() __attribute__((annotate("withargfunc", "os", 22))) {
+}
+
+// BEFORE: cir.global  external @a = #cir.ptr<null> : !cir.ptr<!cir.double>
+// BEFORE-SAME: [#cir.annotation<name = <"withargs" : !cir.array<!s8i x 8>>,
+// BEFORE-SAME: unit = <"[[PATH:.*]]attribute-annotate-multiple.cpp" : 
+// BEFORE-SAME: !cir.array<!s8i x [[FILENAME_LEN:[0-9]+]]>>, 
+// BEFORE-SAME: lineno = <5>, args = #cir.const_struct<{#cir.global_view<@".str">
```

Comment:
Like we discussed offline, at this level we are mapping the annotation, not the source language, `#cir.const_struct` should be used for things coming from the language. We also talked about not creating extra globals at this point, all we need to do here is to store the string again "withargs", etc. The idea here is to bypass most of the information, mapping it to CIR specific types won't help with anything either.